src/player.cpp

```diff
@@ -3551,9 +3551,9 @@ void player::on_hurt( Creature *source, bool disturb /*= true*/ )
         }
         if( !is_npc() ) {
             if( source != nullptr ) {
-                g->cancel_activity_query(_("You were attacked by %s!"), source->disp_name().c_str());
-            } else {
-                g->cancel_activity_query(_("You were hurt!"));
+                g->cancel_activity_query( _( "You were attacked by %s!" ), source->disp_name().c_str() );
+            } else if( !is_elec_immune() ) {
```

Comment:
How does this know the player is *actually* struck by electricity? This only checks whether they are immune to it, but not whether the source of the hurting has anything to do with electricity.